src/libsync/owncloudpropagator.cpp

```diff
@@ -94,6 +94,16 @@ int OwncloudPropagator::hardMaximumActiveJob()
     return max;
 }
 
+PropagateItemJob::~PropagateItemJob()
+{
+    if (auto p = propagator()) {
```

Comment:
Well, it's still undefined behavior to static_cast to the Propagator if we are called from the QObject destructor.